### PR TITLE
add initial implementation of dense Cohere encoder

### PR DESCRIPTION
**Pull request by DosticJelena**
## Problem

Integrating Cohere embedding models.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Describe specific steps for validating this change.

### CODE REVIEW

**Review by acatav — CHANGES_REQUESTED** [review]
LGTM. Only two comments:
1. Can you please add unit tests same as we have for OpenAI?
2. This PR can be merged only after the pinecone text PR is merged and version is bumped

**Comment by igiloh-pinecone** [comment]
Replaced by #203

**Pull request closed without merging**
